contracts/contracts/RequestManager.sol

```diff
@@ -198,30 +216,66 @@ contract RequestManager is Ownable, LpWhitelist, RestrictedCalls, Pausable {
     /// Maps ERC20 token address to tokens
     mapping(address tokenAddress => Token) public tokens;
 
+    /// Compute the minimum liquidity provider fee that needs to be paid for a token transfer.
+    function minLpFee(
+        uint256 targetChainId,
+        address tokenAddress
+    ) public view returns (uint256) {
+        Token storage token = tokens[tokenAddress];
+        Chain storage sourceChain = chains[block.chainid];
+        Chain storage targetChain = chains[targetChainId];
+
+        // The shift by 30 decimals comes from a multiplication of two PPM divisions (1e6 each)
+        // and the 18 decimals division for ether
+        return
+            (((1_000_000 - sourceChain.targetWeightPPM) *
+                sourceChain.transferCost +
+                targetChain.targetWeightPPM *
+                targetChain.transferCost) *
+                (minFeePPM + 1_000_000) *
+                token.ethInToken) / 10 ** 30;
+    }
+
     /// Compute the liquidity provider fee that needs to be paid for a given transfer amount.
     function lpFee(
+        uint256 targetChainId,
         address tokenAddress,
         uint256 amount
     ) public view returns (uint256) {
-        Token storage token = tokens[tokenAddress];
-        return Math.max(token.minLpFee, (amount * token.lpFeePPM) / 1_000_000);
+        uint256 minFee = minLpFee(targetChainId, tokenAddress);
+        return Math.max(minFee, (amount * lpFeePPM) / 1_000_000);
     }
 
     /// Compute the protocol fee that needs to be paid for a given transfer amount.
-    function protocolFee(
-        address tokenAddress,
-        uint256 amount
-    ) public view returns (uint256) {
-        return (amount * tokens[tokenAddress].protocolFeePPM) / 1_000_000;
+    function protocolFee(uint256 amount) public view returns (uint256) {
+        return (amount * protocolFeePPM) / 1_000_000;
     }
 
     /// Compute the total fee that needs to be paid for a given transfer amount.
     /// The total fee is the sum of the liquidity provider fee and the protocol fee.
     function totalFee(
+        uint256 targetChainId,
         address tokenAddress,
         uint256 amount
     ) public view returns (uint256) {
-        return lpFee(tokenAddress, amount) + protocolFee(tokenAddress, amount);
+        return lpFee(targetChainId, tokenAddress, amount) + protocolFee(amount);
+    }
+
+    function transferableAmount(
+        uint256 targetChainId,
+        address tokenAddress,
+        uint256 amount
```

Comment:
is this amount will be actually user total balance ?

Reply:
> is this amount will be actually user total balance ?

Yes, I assume our frontend will use this function when the user presses the "max" button.

Reply:
totalFee can be applied with any value, namely the value the user chooses to send